libcst/metadata/scope_provider.py

```diff
@@ -699,22 +700,66 @@ def visit_Attribute(self, node: cst.Attribute) -> Optional[bool]:
 
     def visit_Call(self, node: cst.Call) -> Optional[bool]:
         self.__top_level_attribute_stack.append(None)
+        if any(
+            qn.name == "typing.TypeVar"
+            for qn in self.scope.get_qualified_names_for(node)
+        ):
+            node.func.visit(self)
+            for arg in node.args[:1]:
+                arg.visit(self)
+            self.__in_annotation.add(node)
+            for arg in node.args[1:]:
+                arg.visit(self)
+            return False
```

Comment:
Why do we visit the `args` twice?
The first parameter of TypeVar is just a name of the defined type.
We should only analyze the other parameters (`args[1:]`).

Reply:
The first argument of `typing.TypeVar` is the name, and is always a non-typed string, the following arguments are restriction types either as positional arguments or as `bound=`.

Reply:
Oh, got it! yes... we can avoid analyzing the first argument, you're right!